src/conditions.lisp

```diff
@@ -47,12 +47,33 @@ PRINT-OBJECT method defined on the class named by NAME."
     `(let ((,method (find-method #'print-object '() '(,name t) nil)))
        (when ,method (remove-method #'print-object ,method)))))
 
+(defmacro define-condition-internal (name (&rest supertypes) direct-slots &rest options)
+  "Defines a new condition type via DEFCLASS, handling the :REPORT options via
+defining a PRINT-object method on the newly created class.
+This is for internal use, allowing any SUPERTYPES."
+  (let* ((report-option (find :report options :key #'car))
+         (other-options (remove report-option options))
+         (supertypes (or supertypes '(condition))))
+    `(progn (defclass ,name ,supertypes ,direct-slots ,@other-options)
+            ,@(if report-option
+                  `(,(expand-define-condition-report-method name report-option))
+                  `(,(expand-define-condition-remove-report-method name)))
+            ',name)))
+
+(defun coerce-condition-supertypes (supertypes)
+  "Ensures that CONDITION is present in the SUPERTYPES list and that all of SUPERTYPES
+members themselves are subclasses of CONDITION."
+  (let ((only-condition-subtypes (remove-if #'(lambda (x)
+                                                  (not (subtypep x 'condition)))
+                                              supertypes)))
+    (or only-condition-subtypes '(condition))))
```

Comment:
This function does not ensure anything; it's a simple filter that quietly removes non-condition supertypes.

In case when we `(define-condition foo (bar) ...)` where `bar` does not name a condition type, I would expect this form to signal an error rather than silently skip `bar`. This issue will be more pronounced with multiple condition supertypes.

Could you define this behavior (it should signal a compile-time warning and/or a runtime error) by adding appropriate test cases to the test suite?

Reply:
"This function does not ensure anything;"
Yes, I'm sorry for the docstring, I wrote before completing the changes and forget to change. Yes, it would quietly remove.

I'll work on the required changes. Thanks for the feedback.